File: 11 opencv（二）/test20210119/prespective.py
import cv2
import numpy as np

img = cv2.imread("2.jpg")

pts1 = np.float32([[25, 30], [179, 25], [12, 188], [189, 190]])
pts2 = np.float32([[0, 0], [200, 0], [0, 200], [200, 200]])

M = cv2.getPerspectiveTransform(pts1, pts2)
dst = cv2.warpPerspective(img, M, (200, 200))

cv2.imshow("src", img)
cv2.imshow("dst", dst)

cv2.waitKey(0)

File: 11 opencv（二）/prespective.cpp
#include<opencv2/opencv.hpp>

using namespace std;

int main() {
    cv::Mat img = cv::imread("2.jpg");

    cv::Point2f pts1[] = {cv::Point2f(25, 30), cv::Point2f(179, 25), cv::Point2f(12, 188), cv::Point2f(189, 190)};
    cv::Point2f pts2[] = {cv::Point2f(0, 0), cv::Point2f(200, 0), cv::Point2f(0, 200), cv::Point2f(200, 200)};

    cv::Mat M = cv::getPerspectiveTransform(pts1, pts2);

    cv::Mat dst_img;
    cv::warpPerspective(img, dst_img, M, img.size());

    cv::imshow("src", img);
    cv::imshow("dst", dst_img);
    cv::waitKey(0);
}


